CMakeLists.txt

```diff
@@ -1656,6 +1656,10 @@ if(CONFIG_OUTPUT_STAT)
 endif()
 
 if(CONFIG_BUILD_OUTPUT_STRIPPED)
+if(BOARD MATCHES "^native_posix")
+# Using a random toolchain makes zephyr.strip not interesting anyway.
+message("WARNING: not stripping ${BOARD} because of bug #51821")
```

Comment:
This is not an acceptable way to do this.

If there is a need for not supporting a feature on a board / arch, then that should be ensured using Kconfig.

But for issue #51821 as fix is ready here: #51837

Reply:
Fix merged, workaround gone - resolved.